decidim-bulletin_board-ruby/lib/decidim/bulletin_board/client.rb

```diff
@@ -34,6 +34,11 @@ def configured?
         private_key && server && api_key
       end
 
+      def setup_election(election_data)
+        message_id = "#{election_data[:election_id]}.create_election+a.#{authority_slug}"
+        Decidim::BulletinBoard::CreateElection.call(election_data, message_id)
```

Comment:
The `election_data` has a `:message_id` entry, so maybe we could get it directly from there to be more DRY.
Actually, we have discussed about this with @slickepinne, and we reached the conclusion that we should move the JSON building code to the gem (in the future). 
After doing that, I don't have a clear idea about where is the best place to calculate the `message_id`. But I'd say that the responsible should be the `CreateElection` class, maybe using a concern to reuse the common part of all message identifiers. What do you think?

Reply:
Yeah! We had the same discussion 😄 We should definetly move it to the gem as well. About the `message_id` generation I'm not sure and I had the same problem with the JS client. I put it in the `KeyCeremony` class for the moment but doesn't feel right.

I think that it will be much clear when we start implementing the EG wrapper and dealing with different messages maybe. For the Ruby part maybe it's better to have it in the `CreateElection` class as you said. We can do that when we refactor the whole `election_data` stuff 👍